src/components/Swap/Toolbar/Caption.tsx

```diff
@@ -28,20 +29,32 @@ const CaptionRow = styled(Row)<{ gap: number; shrink?: number }>`
   height: 100%;
 `
 
+interface CaptionTooltip {
+  content: ReactNode
+  placement?: Placement
+}
+
 interface CaptionProps {
   icon?: Icon
   caption: ReactNode
   color?: Color
+  tooltip?: CaptionTooltip
 }
 
 interface GasEstimateProps {
   gasUseEstimateUSD: CurrencyAmount<Token> | undefined
 }
 
-function Caption({ icon: Icon = AlertTriangle, caption, color = 'secondary' }: CaptionProps) {
+function Caption({ icon: Icon = AlertTriangle, caption, color = 'secondary', tooltip }: CaptionProps) {
   return (
-    <CaptionRow gap={0.5}>
-      <LargeIcon icon={Icon} color={color} />
+    <CaptionRow gap={0.5} shrink={0}>
+      {Boolean(tooltip) ? (
```

Comment:
This should not need to be wrapped in a Boolean, because the tooltip will never be returned - it is a ternary.